src/main/java/ch/njol/skript/bukkitutil/ItemUtils.java

```diff
@@ -379,4 +380,42 @@ public static boolean isGlass(Material material) {
 				return false;
 		}
 	}
+
+	/**
+	 * Applies a provided {@code Consumer} to the meta of the provided {@code ItemStack} and returns the updated {@code ItemStack} (with updated {@code ItemMeta}).
+	 *
+	 * @param itemStack the item whose meta is to be changed using the provided Consumer
+	 * @param metaChanger a consumer to update the meta of the provided ItemStack
+	 * @param <T>
+	 * @return the updated item
+	 */
+	public static <T extends ItemMeta> T changeItemMeta(@NotNull ItemStack itemStack, @NotNull Consumer<T> metaChanger) {
```

Comment:
Paper has these `editMeta` methods which might be useful call here if available: https://jd.papermc.io/paper/1.21.4/org/bukkit/inventory/ItemStack.html#editMeta(java.util.function.Consumer)

Though, I'm not sure they are doing anything different internally from what is being done here, so it might be worth looking into that first.

Reply:
As far as I remember they did the exact same thing as he does here, when he was working on it I mentioned in a discord call about it.
![image](https://github.com/user-attachments/assets/582491d3-2413-425a-96af-0f21918447c3)

Reply:
Ah yep, if it's the same then no need to worry about it.